kubectl/upload/util.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	return os.Getenv("USERPROFILE") // windows
+}
+
+func calculateMD5(path string) (string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hash := md5.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%x", hash.Sum(nil)), nil
+}
+
+func tarGz(src, dst string) error {
+	tarFile, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer tarFile.Close()
+
+	gzWriter := gzip.NewWriter(tarFile)
+	defer gzWriter.Close()
+
+	tarWriter := tar.NewWriter(gzWriter)
+	defer tarWriter.Close()
+
+	return filepath.Walk(src, func(path string, info os.FileInfo, err error) error {
```

Comment:
Add comment/issue to respect `.dockerignore` (can happen later).

Reply:
I would think the build job itself would respect the `.dockerignore` but not compressing/uploading files in `.dockerignore` would be an improvement. 👍